### PR TITLE
Fix add group button

### LINKED ISSUE #27435: Create group doesn't work (22.0.0 beta3)
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Go to Users menu
2. Click button "+ Add Group"
3. Nothing happens

### Expected behaviour
NC should prompt me for the group's name

### Actual behaviour
Nothing happens

### Additionnal info

Creating the group using occ works well

### Server configuration

**Operating system:** Debian 

**Web server:** Apache 

**Database:** Mysql

**PHP version:** 7.3

**Nextcloud version:** master

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

Can't get the result, page returns "Integrity checker has been disabled. Integrity cannot be verified." even if I put `'integrity.check.disabled' => false` or `'integrity_check_disabled' => false,` in config.php. Another bug?
```

**List of activated apps:**
<details>
<summary>
Enabled:
  - accessibility: 1.7.0
  - calendar: 2.2.2
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.18.0
  - deck: 2.0.0-alpha1
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_sharing: 1.13.2
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - groupfolders: 10.0.2
  - lookup_server_connector: 1.9.0
  - mail: 1.10.0-alpha.6
  - notestutorial: 18.0.0
  - oauth2: 1.9.0
  - provisioning_api: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - systemtags: 1.11.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.1
  - updatenotification: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.6.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
  - workspace: 0.0.10
Disabled:
  - admin_audit
  - bruteforcesettings
  - circles
  - encryption
  - files_external
  - files_texteditor
  - impersonate
  - testing
  - user_ldap
</summary>
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud-dev.bollu.be",
            "192.168.0.128",
            "127.0.0.1"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "22.0.0.6",
        "overwrite.cli.url": "http:\/\/nextcloud-dev.bollu.be",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "log_type": "syslog",
        "logfile": "",
        "loglevel": 0,
        "maintenance": false,
        "debug": true,
        "allow_local_remote_servers": true,
        "integrity.check.disabled": false
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox latest 89.0 64bits

**Operating system:** Windows 10

### Logs

There is no single log. Nothing happens when I click on the button: The browser's console doesn't output anything and no request is sent to the server.

### LINKED ISSUE #27866: NC 22, "Add group" Button just doing nothing
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Upgrade or install NextCloud v22
2. Become an Admin or Groupd-Admin and navigate to "Users"
3. Click on "Add group"

### Expected behaviour
The "Add group" element should turn into a text input element and prompt for the new group name

### Actual behaviour
NOTHING, the "Add group" element does not turn into a text input element

### Server configuration

**Operating system:**
Linux Ubunt Focal, kernel 5.12.9

**Web server:**
Apache2

**Database:**
mariadb, as installed on Ubuntu Focal

**PHP version:**
PHP 8.0.8, ondrej PPA "deb http://ppa.launchpad.net/ondrej/php/ubuntu focal main"

**Nextcloud version:** (see Nextcloud admin page)
v22, just upgrade to the "beta" channel

**Updated from an older Nextcloud/ownCloud or fresh install:**
v21, after applying the latest update

**Where did you install Nextcloud from:**
YOUR SERVER

**Signing status:**
<details>
<summary>Signing status</summary>

What do you mean by "Signing status"????? This question is just confusing.

I SKIP THE REMAINING QUESTIONS OF THIS QUESTIONAIR. THIS IS JUST TOO MUCH.

"Add Group" is just broken, doing nothing in V22. Full stop.

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

### PR DESCRIPTION
**Pull request by kesselb**
Fix #27435 
Fix https://github.com/nextcloud/server/pull/27866

https://user-images.githubusercontent.com/3902676/124805637-11969a80-df5c-11eb-983e-35c1bccf902b.mp4

### CODE REVIEW

**Review by LukasReschke — APPROVED** [review]
🚀 🥳 🎉

File: apps/settings/src/views/Users.vue [hunk 1]
```diff
@@ -445,51 +445,40 @@ export default {
...
+			this.$refs.addGroup.editingActive = true
+			this.$refs.addGroup.onMenuToggle(false)
+			this.$nextTick(() => {
+				this.$refs.addGroup.$refs.editingInput.focusInput()
 			})
 		},
 
+		hideAddGroupForm() {
+			this.$refs.addGroup.editingActive = false
+			this.$refs.addGroup.editingValue = ''
```
Comment (Pytal) [inline apps/settings/src/views/Users.vue:479]:
```suggestion
			this.$refs.addGroup?.editingActive = true
			this.$refs.addGroup?.onMenuToggle(false)
			this.$nextTick(() => {
				this.$refs.addGroup?.$refs?.editingInput?.focusInput()
			})
		},

		hideAddGroupForm() {
			this.$refs.addGroup?.editingActive = false
			this.$refs.addGroup?.editingValue = ''
```
Some optional chaining safety :)

**Review by Pytal — CHANGES_REQUESTED** [review]

File: apps/settings/src/views/Users.vue (hunk 1 above)
Comment (kesselb) [reply apps/settings/src/views/Users.vue:479]: Hmm. Can $refs.addGroup be null? I would argue that something is broken if addGroup is null ;)

Comment (Pytal) [reply apps/settings/src/views/Users.vue:479]: good point 🤔 hadn't looked over all of it yet, refs here won't be null when the function is called :)

**Review by Pytal — APPROVED** [review]

File: apps/settings/src/views/Users.vue (hunk 1 above)
Comment (nickvergessen) [reply apps/settings/src/views/Users.vue:479]: $ref is null when triggered before mounted or mounted finished, don't recall atm.

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Please rebase and compile, you need to remove the extra settings files
```
	deleted:    apps/settings/js/vue-settings-users-7f4f43139d7b1af5f465.js
	deleted:    apps/settings/js/vue-settings-users-7f4f43139d7b1af5f465.js.map

```

**Comment by kesselb** [comment]
> Please rebase and compile, you need to remove the extra settings files
> 
> ```
> 	deleted:    apps/settings/js/vue-settings-users-7f4f43139d7b1af5f465.js
> 	deleted:    apps/settings/js/vue-settings-users-7f4f43139d7b1af5f465.js.map
> ```

Done

**Pull request merged**